### DISCUSSION TITLE
[j4] We can't add a pdf in content ... ?

### DISCUSSION THREAD

**Discussion by micker** [joomla/joomla-cms#32642] [General] [answered]
hello i am testing j4 with new media manager with end-user
the new media manager is realy cool but for now only for image
adding a simple pdf to link it in item is a nighmare
very confusing for end user
![image](https://user-images.githubusercontent.com/2539440/98301090-8b841780-1fba-11eb-88df-7e6022f41a74.png)
i think we can have 2 solutions
=> rename button to add image => add media, and change title of popup and allow to load file link
=> adding a new link in dropdown
![image](https://user-images.githubusercontent.com/2539440/98301402-fa617080-1fba-11eb-87d5-9671293d3fb5.png)

other return, its realy strange to have cms content to load file or image ... not user friendly
any plan for this ?

**Comment by micker** [comment]
hi nobody hurt by this ? or a plan is already on road ?

**Comment by micker** [comment]
i try to change the title i can't imagine that in 2021 we can say that we have a new media manager and we can't add pdf or file easier in content ...

**Comment by sluc69** [comment]
I guess the developers are not users of Joomla. The media manager J4 is the same as in J3; basic and limited. Perhaps they let the add-on developers come up with a better version.

**Comment by sozzled** [comment]
> The new media manager is realy cool but for now only for image
> adding a simple pdf to link it in item is a nighmare
> very confusing for end user

Took me 10 seconds to upload a .pdf file using J! 4 Media Manager.  Took me a few seconds longer to figure out how to get the link.  It's not intuitive but it works.  I wouldn't describe it as a nightmare.  It works.

I don't like J! 4 Media Manager as much as I was happy with J! 3 Media Manager but that's just personal preference.  The market will decide if people like the new backend administration.  I like the fact that in J! 4, you can rename (or download) files with the Media Manager ...

**Comment by drmenzelit** [comment]
There is a GSoC student (and a team of mentors) working right now on improvements for the media manager

**Comment by sluc69** [comment]
Is there a reason why media manager is using HTTP DELETE method for removing items?
J!3 did not use this method. I have a 'LimitExcept GET POST HEAD' on the webserver (only except GET POST HEAD methods) and that works fine with J!3.
![Screenshot 2021-06-16 110630](https://user-images.githubusercontent.com/82952282/122151935-05567a80-ce93-11eb-9392-42332fc8de04.jpg)

**Reply by wilsonge** [reply]
Yes. It using a proper REST API for the backend. The API that ships in Joomla 4 will also require DELETE (and for the API specifically PATCH & PUT would be required too)

**Comment by micker** [comment]
but we can't wait a goc student to add this simply function
in j4 we can't add a pdf in content
![image](https://user-images.githubusercontent.com/2539440/122171549-206bbf00-ce80-11eb-80f8-a8ec9f3e70df.png)
in j3 we can... its an important regression
![image](https://user-images.githubusercontent.com/2539440/122171852-793b5780-ce80-11eb-8157-dbecf5bcbc57.png)

@wilsonge and @jteam ?

**Reply by brianteeman** [reply]
Sorry I dont understand what your screenshot is showing

**Reply by Fedik** [reply]
@micker in joomla 3 it also not possible, 
your screenshot showing totally unrelated things

**Reply by micker** [reply]
sorry @Fedik you are right my screenshot isn't good
but can we imagine to publish a new cms and say that we have a new media manger and can't add a file easier in content ? in 2021
i didn't want to hurt anyone but i didn't think that we can forgot this to simple thing

**Reply by brianteeman** [reply]
Where is the regression

**Reply by micker** [reply]
joomla project said that we have a nive mediamanager for j4 but at end we only offer insert image in content realy ??? rename it to image manager.
ok we can add image pdf and more in media manager if we can't add it in content no sense... people say that joomla! is difficult, not realy for me but end customer (person how create content) can't do simple action ... i don't understand why the team doesn't want to add this ??? what is the goal if dev works many weeks on import sample, nobody realy use (only for demo) it but adding a simple media content like a pdf its a dayli function ... every body need it !
i know team do a lot of effort on tinymce its realy great tool but we can't just say change it for jce only for this

**Reply by brianteeman** [reply]
> every body need it !

That is a personal opinion and experience. For example there are no pdf on my sites

**Reply by micker** [reply]
yes but in your clients sites ... i can't imagine that nobody need this lol

**Reply by brianteeman** [reply]
when I say "my sites" I mean all the sites I have built. I am not saying it would be useful just disputing that every site need it

**Reply by micker** [reply]
the qustion isn't if me or you need this but if many user need it
if we check other important CMS have this function out of box, its dommage if joomla! forgot it for his major release

**Reply by sluc69** [reply]
How would this PDF be displayed? As a link (opens in browser or pdf reader) or  embedded in the page?

**Reply by Fedik** [reply]
As a link

**Comment by rinka88** [comment]
Cool media manager for joomla 3 is a quantum manager from norrnext

**Comment by mrownicki** [comment]
Not only pdf, docx xls, etc.. Joomla don't have anything to upload and linking files. 
That's why JCE is popular.
Someone once wrote to me that I should use JCE because tinymce is limited and it will be so. I don't remember who but one of the developers team.

btw. IN first alpha there was a button to add files, but why is it gone ??

This is 2021 .. CMS is not just for writing notes ...

**Comment by sozzled** [comment]
I don't understand half of this discussion; a lot of it is off-topic IMO.  I replied with my understanding of the original "problem" [here](https://github.com/joomla/joomla-cms/discussions/32642#discussioncomment-849078).

I don't know how much "easier" the Media Manager can be ... :question:  :question:  :question:

**Reply by mrownicki** [reply]
> Took me 10 seconds to upload a .pdf file using J! 4 Media Manager. Took me a few seconds longer to figure out how to get the link. It's not intuitive but it works. I wouldn't describe it as a nightmare. It works.

I think you're not newbie user this operation must be simple as can. 
Button add files -> upload files -> add files to article as a link. 3 step. That's all every user wants.

You probably won't be sending every user to change and add the supported file type to the media manager.

Therefore, a WP where I drag and drop the file for me and has 60% of the market .. Because in Joomla everyone must first read a pile of documentation for such a trivial operation.

Seriously people this is sad..
When a good change is proposed every time, no one pays attention to it or immediately says it is possible, but you have to take 20 steps. So there is no need to change because, after all, there is functionality.

**Comment by dgrammatiko** [comment]
Implemented here: https://github.com/joomla/joomla-cms/pull/34634

**Reply by micker** [reply]
thanks RTC @dgrammatiko  !!!

**Comment by micker** [comment, marked as answer]
https://github.com/joomla/joomla-cms/pull/34634
